### ISSUE TITLE
"Module 'ZendeskProviderSDK' not found" error when ZendeskSDK is added as second-level CocoaPod

### ISSUE THREAD

**Issue by davidjed** [zendesk/zendesk_sdk_ios#268]
This on Xcode 8, iOS 10, Zendesk 1.7.6.1. Occurs when ZendeskSDK is added as a second-level CocoaPod (i.e. it's a CocoaPod of ProjectA, and ProjectA is a CocoaPod of ProjectB; ProjectB is where the error occurs). ZendeskSDK can be added and implemented with no problems in ProjectA and ProjectB directly, but adding the extra layer causes this error. Not sure if Podfile/Podspec require certain flags or settings to prevent this.

**Comment by davidjed** [comment]
To repro this, add the Zendesk SDK to any project, then add that project to another project. The next-level descendent project is where the error occurs.

**Comment by StevenDiviney** [comment]
@davidjed this sounds like expected behavior to me. A package manager doesn't typically bundle all of the dependencies in with a package. ZendeskSDK should be listed as a dependency of every project that needs it. If you need to ensure the same version of the SDK is used in multiple projects cocoapods allows you to do so.

**Comment by davidjed** [comment]
Oh yeah I should have clarified... I am adding it to both projects (in the example above ProjectA and ProjectB) and still getting the error.

**Comment by StevenDiviney** [comment]
That changes things. I'll follow this up in the ticket you opened as I want to have a look at your podfiles.

**Comment by davidjed** [comment]
OK sounds good -- here are the Podfiles for both projects; the root project, ios-common, and the top-level project, ReceiptHog (I gave them both a project name suffix and .txt file suffix to allow them to be attached here). BTW, this error happens even if I add ZendeskSDK to ios-common's podspec as a dependency and remove it from ReceiptHog's Podfile, or leave it in that Podfile. I tried all those combinations and in between doing so also blew away the DerivedData directory for the project that Xcode creates as sometimes the artifacts in that directory can mess with subsequent builds and produce errors even after doing a "clean" in Xcode.

Let me know what you come up with... thanks!


[Podfile-ioscommon.txt](https://github.com/zendesk/zendesk_sdk_ios/files/717376/Podfile-ioscommon.txt)
[Podfile-receipthog.txt](https://github.com/zendesk/zendesk_sdk_ios/files/717377/Podfile-receipthog.txt)
